code/core/src/preview-api/modules/store/StoryStore.ts

```diff
@@ -323,7 +323,15 @@ export class StoryStore<TRenderer extends Renderer> {
             }
             return Object.assign(storyAcc, { [key]: value });
           },
-          { args: story.initialArgs }
+          {
+            //
+            args: story.initialArgs,
+            globals: {
+              ...this.userGlobals.initialGlobals,
+              ...this.userGlobals.globals,
```

Comment:
Does it make sense to include "temporary" global changes in extract? 

For instance if I go to http://localhost:6006 should I get a different result for `.extract()` than if I go to http://localhost:6006?globals:X=y ?